cabal-install/Distribution/Client/CmdInstall.hs

```diff
@@ -477,7 +477,7 @@ installAction (configFlags, configExFlags, installFlags, haddockFlags, testFlags
       $ defaultProgramDb
 
   (compiler@Compiler { compilerId =
-    compilerId@(CompilerId compilerFlavor compilerVersion) }, platform, progDb') <-
+    compilerId@(CompilerId compilerFlavor compilerVersion) }, platform, progDb) <-
```

Comment:
If this shadows, then it will cause `-Wall` warning, won't it?

```
Distribution/Client/CmdInstall.hs:480:73: warning: [-Wname-shadowing]
    This binding for ‘progDb’ shadows the existing binding
      bound at Distribution/Client/CmdInstall.hs:471:5
<no location info>: error: 
Failing due to -Werror.
```

I'd rename original `progDb` to e.g. `preProgDb` and adding some comments how they differ

```
-- ProgramDb with directly user specified paths
preProgDb = ...

-- progDb is a program database with compiler tools configured properly
(compiler@ ... progDb) <- ...
```

Reply:
Yeah, I wasn't sure if we have -Wall and -Werror on by default even.

Honestly I vehemently disagree with the `-Wname-shadowing` warning. In fact in my opinion because this warning discourages shadowing names in do blocks like it does was _the_ reason for this bug to even get introduced in the first place. People are just used to imperative style of doing things and especially in a do block that should just be something Haskell lets you do. I digress -- should write a GHC proposal about that instead of ranting ;)